### ISSUE TITLE
Current weekly-test-build (on snap) broken/unplayable

### ISSUE THREAD

**Issue by agrrr3** [freeorion/freeorion#3326]
Bug Report
==========

I just built and published freeorion_2021-02-23.4d6870e rev=168 channel=edge.
Doing a smoke test game, starting a new game leaves all meters zero. Homeplanet has no population. Ships have no fuel.

**Comment by agrrr3** [comment]
in the logs i did only find the new [error] / [warning] entries in AI.log and freeoriond.log
```
freeoriond.log:17:12:16.772689 {0x00007fbf68bc4780} [error] python : warnings.py:30 : _frozen_importlib:219: RuntimeWarning: to-Python converter for std::pair<MeterType, std::vector<Effect::AccountingInfo, std::allocator<Effect::AccountingInfo> > > already registered; second conversion method ignored.
freeoriond.log:17:12:16.772980 {0x00007fbf68bc4780} [error] python : warnings.py:30 : _frozen_importlib:219: RuntimeWarning: to-Python converter for std::pair<int const, boost::container::flat_map<MeterType, std::vector<Effect::AccountingInfo, std::allocator<Effect::AccountingInfo> >, std::less<MeterType>, boost::container::new_allocator<std::pair<MeterType, std::vector<Effect::AccountingInfo, std::allocator<Effect::AccountingInfo> > > > > > already registered; second conversion method ignored.
```

**Comment by geoffthemedio** [comment]
> in the logs i did only find the new [error] / [warning] entries in AI.log and freeoriond.log

Meaning of this is unclear. Post the logs, probably at trace level for effects and server general.

**Comment by agrrr3** [comment]
i meant i didnt see anything suspicious on error/warning level

[logs.zip](https://github.com/freeorion/freeorion/files/6038882/logs.zip)

**Comment by o01eg** [comment]
I'm writing parser tests for tech and in commit 04373221f59dff9c8228d65f3b1b18c0d8ddb039 got empty operands in `And` in scope of `CON_ORBITAL_HAB` tech.

**Comment by geoffthemedio** [comment]
> I'm writing parser tests for tech and in commit [0437322](https://github.com/freeorion/freeorion/commit/04373221f59dff9c8228d65f3b1b18c0d8ddb039) got empty operands in `And` in scope of `CON_ORBITAL_HAB` tech.

That commit shouldn't change the parsers that affect the relevant conditions...
````
            scope = And [
                Species
                OwnedBy empire = Source.Owner
            ]
````
Perhaps also add a test for just `Species` and `OwnedBy empire = Source.Owner` and perhaps also just `Source.Owner` ?

**Comment by o01eg** [comment]
I don't have neither species nor empires initialized for test. Maybe it somehow related.

**Comment by geoffthemedio** [comment]
> I don't have neither species nor empires initialized for test. Maybe it somehow related.

I don't think constructing a Condition or ValueRef requires accessing the SpeciesManager or any ScriptingContext stuff, unless there is a ValueRef Operation that needs to cache a constant value, which this doesn't. Even then, it wouldn't access the SpeciesManager, though it might fail when trying to construct the ScriptingContext.

**Comment by agrrr3** [comment]
Thanks to the tests made by o01eg and digging a bit more into snapcraft build i found a way almost efficient manual bisection (from commandline inside the multipass build VM, skipping installing as i do not need UI). Automation was not possible as the other smoke test always fails.

@geoffthemedio  it seems my boost 1.65 does not work correctly with custom thread pool implementation for pre-Boost 1.66

a94336432d037562b087ef7b628e03952490070a works. The next is broken:

commit e540f459a925fb35af30a52c5fbfd14a9f6fb2e0
Author: geoffthemedio <geoffthemedio@gmail.com>
Date:   Sun Feb 21 13:14:54 2021 +0100

    add a custom thread pool implementation for pre-Boost 1.66


i will try if a ubuntu 20.4 (core20) based build with a different boost works.

**Comment by geoffthemedio** [comment]
https://github.com/freeorion/freeorion/commit/f6c51e38e5b7bbb13624a905a315533ef2ef84ec

**Comment by agrrr3** [comment]
Made a core20 build with libboost*1.67.0 (master from 2021-02-23, 2b7418). 
Still unexpectedly broken. Will try f6c51e3 next

**Comment by agrrr3** [comment]
> Made a core20 build with libboost*1.67.0 (master from 2021-02-23, 2b7418).
> Still unexpectedly broken. Will try [f6c51e3](https://github.com/freeorion/freeorion/commit/f6c51e38e5b7bbb13624a905a315533ef2ef84ec) next

built the current master baf355fa which contains f6c51e3. the revert did not fix the issue (at least with core20/boost 1.67). not sure whats going on or how to proceed best.

**Comment by agrrr3** [comment]
> > Made a core20 build with libboost*1.67.0 (master from 2021-02-23, 2b7418).
> > Still unexpectedly broken. Will try [f6c51e3](https://github.com/freeorion/freeorion/commit/f6c51e38e5b7bbb13624a905a315533ef2ef84ec) next
> 
> built the current master [baf355f](https://github.com/freeorion/freeorion/commit/baf355faf3d43d1a132001b522fbc1e84c8bf348) which contains [f6c51e3](https://github.com/freeorion/freeorion/commit/f6c51e38e5b7bbb13624a905a315533ef2ef84ec). the revert did not fix the issue (at least with core20/boost 1.67). not sure whats going on or how to proceed best.

probably my build environment is contaminated. the resulting snap claims to be d070fcd which is my changes on top of 2b747 (Feb 23rd). also both libboost1.65.1 and 1.67.0 are included.

**Comment by agrrr3** [comment]
Ok, I cleaned and rebuild (and fixed some). The core20 build now works with the latest master. So probably the boost issue was the cause. (didnt have time to check other combination)
